server/src/test/java/org/elasticsearch/cluster/coordination/DeterministicTaskQueue.java

```diff
@@ -0,0 +1,361 @@
+            public long relativeTimeInMillis() {
+                return currentTimeMillis;
+            }
+
+            @Override
+            public long absoluteTimeInMillis() {
+                return currentTimeMillis;
+            }
+
+            @Override
+            public Counter estimatedTimeInMillisCounter() {
+                return new Counter() {
+                    @Override
+                    public long addAndGet(long delta) {
+                        throw new UnsupportedOperationException();
+                    }
+
+                    @Override
+                    public long get() {
+                        return currentTimeMillis;
+                    }
+                };
+            }
+
+            @Override
+            public ThreadPoolInfo info() {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public Info info(String name) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public ThreadPoolStats stats() {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public ExecutorService generic() {
+                return getExecutorService();
+            }
+
+            @Override
+            public ExecutorService executor(String name) {
+                return getExecutorService();
+            }
+
+            @Override
+            public ScheduledFuture<?> schedule(TimeValue delay, String executor, Runnable command) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public Cancellable scheduleWithFixedDelay(Runnable command, TimeValue interval, String executor) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public Runnable preserveContext(Runnable command) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void shutdown() {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void shutdownNow() {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public boolean awaitTermination(long timeout, TimeUnit unit) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public ScheduledExecutorService scheduler() {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void close() {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public ThreadContext getThreadContext() {
+                throw new UnsupportedOperationException();
+            }
+        };
+    }
+
+    private class DeferredTask {
```

Comment:
can this be a static class?

Reply:
It used `currentTimeMillis` but on closer inspection this was just in a vacuous assertion.